sched/scheduler/stateful_scheduler.go

```diff
@@ -406,25 +424,100 @@ func (s *statefulScheduler) step() {
 	s.killJobs()
 	s.scheduleTasks()
 
-	remaining := 0
-	waitingToStart := 0
+	s.updateStats()
+}
+
+/*
+ * update the stats monitoring values:
+ * number of job requests running or waiting to start
+ * number of jobs waiting to start
+ * number of tasks currently running
+ * total number of waiting or running tasks
+ *
+ * for each unique requestor count:
+ * . number of tasks running
+ * . number of tasks waiting
+ * . number of jobs running or waiting to start
+ */
+func (s *statefulScheduler) updateStats() {
+	remainingTasks := 0
+	jobsWaitingToStart := 0
+
+	// reset current counts to 0
+	for _, counts := range s.requestorsCounts {
+		counts["jobsRuning"] = 0
+		counts["jobsWaitingToStart"] = 0
+		counts["numRunningTasks"] = 0
+		counts["numWaitingTasks"] = 0
+	}
+
+	// get job and task counts by requestor, and overall jobs stats
 	for _, job := range s.inProgressJobs {
-		remaining += (len(job.Tasks) - job.TasksCompleted)
+		requestor := job.Job.Def.Requestor
+		if _, ok := s.requestorsCounts[requestor]; !ok {
+			// first time we've seen this requestor, initialize its map entry
+			counts := make(map[string]int)
+			s.requestorsCounts[job.Job.Def.Requestor] = counts
+			counts["jobsWaitingToStart"] = 0
+			counts["jobsRuning"] = 0
+			counts["numRunningTasks"] = 0
+			counts["numWaitingTasks"] = 0
+		}
+
+		remainingTasks += (len(job.Tasks) - job.TasksCompleted)
 		if job.TasksCompleted+job.TasksRunning == 0 {
-			waitingToStart += 1
+			jobsWaitingToStart += 1
+			s.requestorsCounts[requestor]["jobsWaitingToStart"]++
+			s.requestorsCounts[requestor]["numWaitingTasks"] += len(job.Tasks)
+		} else if job.getJobStatus() == sched.InProgress {
+			s.requestorsCounts[requestor]["jobsRuning"]++
```

Comment:
"jobsRuning" -> "jobsRunning" (at least it's consistent!)

Reply:
fixed